src/components/DonorDetails/index.js

```diff
@@ -48,10 +49,10 @@ function DonorDetails(props) {
           <Particulars donorDetails={donorInfo} />
           <Contact donorDetails={donorInfo} />
         </div>
-        <DonorTable donorDetails= {donorInfo} />
+        <DonorTable donorDetails={donorInfo} />
       </div>
     )
-  } else return null
+  } else return <Spin />
```

Comment:
This can look nicer with the `? ... : ...` approach that you are already using in `Dashboard/index.js`